tests/tests-hz38/src/main/java/com/hazelcast/simulator/tests/quorum/QuorumMapTest.java

```
@@ -0,0 +1,101 @@
+
+import org.junit.Assert;
+
+import com.hazelcast.core.IMap;
+import com.hazelcast.quorum.QuorumException;
+import com.hazelcast.simulator.test.AbstractTest;
+import com.hazelcast.simulator.test.BaseThreadState;
+import com.hazelcast.simulator.test.annotations.Setup;
+import com.hazelcast.simulator.test.annotations.TimeStep;
+
+public class QuorumMapTest extends AbstractTest {
+
+    // properties
+    public int keyCount = 100;
+    public int gracePeriodInMilliSec = 2000;
+    private volatile long lastInconsistencyTimestamp = 0L;
+    private IMap<Long, Long> map;
+
+    @Setup
+    @SuppressWarnings("unchecked")
+    public void setup() {
+        map = targetInstance.getMap(name);
+    }
+
+    @TimeStep
+    public void testPut(BaseThreadState state) {
+        final int quorumCount = targetInstance.getConfig()
+                .getQuorumConfig("map-quorum-ref").getSize();
+        boolean operationFailed = false;
+
+        try {
+            map.put((long) state.randomInt(keyCount), 0L);
+            if (targetInstance.getCluster().getMembers().size() < quorumCount) {
+                if (!(lastInconsistencyTimestamp > 0L)) {
+                    startGracePeriodTimer();
+                } else {
+                    if (isGracePeriodElapsed()) {
+                        operationFailed = true;
+                    }
+                }
+            } else {
+                resetGracePeriodTimer();
+            }
+        } catch (QuorumException qe) {
+            if (targetInstance.getCluster().getMembers().size() >= quorumCount) {
+                if (!(lastInconsistencyTimestamp > 0L)) {
+                    startGracePeriodTimer();
+                } else {
+                    if (isGracePeriodElapsed()) {
+                        operationFailed = true;
+                    }
+                }
+            } else {
+                resetGracePeriodTimer();
+            }
+        }
+
+        if (operationFailed) {
+            Assert.fail(String
```

You can just throw an Exception. You don't need to call Assert.fail. Any exception is fine.